core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -901,6 +900,67 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    def handleUnauthorizedRequest = {
+      createTopicRequest.topics.asScala.map { case (topic, _) =>
+        (topic, Errors.CLUSTER_AUTHORIZATION_FAILED)
+      }
+    }
+
+    def handleAuthorizedRequest = {
+      val brokers = metadataCache.getAliveBrokers.map(_.id)
+      createTopicRequest.topics.asScala.map { case (topic, arguments) =>
+        try {
+          // Prep configs
+          val configs = new Properties()
+          arguments.configs.asScala.foreach { case (key, value) =>
+            configs.setProperty(key, value)
+          }
+          LogConfig.validate(configs)
+
+          // Prep assignment
+          val assignments = {
+            if (!arguments.replicasAssignments.isEmpty) {
+              // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
+              // this follows the existing logic in TopicCommand
+              arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
+                (partitionId.intValue, replicas.asScala.map(_.intValue).toSeq)
+              }
+            } else {
+              AdminUtils.assignReplicasToBrokers(brokers, arguments.partitions, arguments.replicationFactor)
+            }
+          }
+
+          // Create the topic
```

Comment:
The fact that we need to add a comment to describe a single method call implies that we are doing something wrong. Does the `AdminUtils` method name need to be so obscure?